### PR TITLE
KYV-603 - feature/KYV-603-list-and-get-merchant

### PR DESCRIPTION
**Pull request by severikupari1**

### CODE REVIEW

**Comment by severikupari1** [comment]
- [ ] merchant create endpoint pr should be merged before this

File: merchantapi/src/main/java/fi/hel/verkkokauppa/merchant/api/dto/MerchantDto.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import javax.validation.constraints.NotBlank;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+@Data
+@ToString
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class MerchantDto {
+    // If given do not create new merchant
+    String merchantId;
```
Comment (martinldev) [inline merchantapi/src/main/java/fi/hel/verkkokauppa/merchant/api/dto/MerchantDto.java:17]: Should these fields be set as private for field access control? Lomboks @Data -annotation will create necessary getters and setters for the values.

File: merchantapi/src/main/java/fi/hel/verkkokauppa/merchant/api/dto/MerchantDto.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import fi.hel.verkkokauppa.merchant.model.ConfigurationModel;
+import lombok.Data;
+import lombok.ToString;
+
+import javax.validation.constraints.NotBlank;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+@Data
+@ToString
```
Comment (martinldev) [inline merchantapi/src/main/java/fi/hel/verkkokauppa/merchant/api/dto/MerchantDto.java:13]: @Data annotation should contain the @ToString generation also (https://projectlombok.org/features/Data) - no need to specify separate @ToString.

File: merchantapi/src/main/java/fi/hel/verkkokauppa/merchant/model/ConfigurationModel.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package fi.hel.verkkokauppa.merchant.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import lombok.Data;
+import lombok.ToString;
+import org.springframework.data.elasticsearch.annotations.Field;
+import org.springframework.data.elasticsearch.annotations.FieldType;
+
+@Data
+@ToString
```
Comment (martinldev) [inline merchantapi/src/main/java/fi/hel/verkkokauppa/merchant/model/ConfigurationModel.java:10]: @DaTa annotation should contain the @tostring generation also (https://projectlombok.org/features/Data) - no need to specify separate @tostring.

File: merchantapi/src/main/java/fi/hel/verkkokauppa/merchant/model/ConfigurationModel.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class ConfigurationModel {
+    @Field(type = FieldType.Text)
+    public String key;
```
Comment (martinldev) [inline merchantapi/src/main/java/fi/hel/verkkokauppa/merchant/model/ConfigurationModel.java:14]: Should these fields be set as private for field access control? Lomboks @DaTa -annotation will create necessary getters and setters for the values.

File: merchantapi/src/main/java/fi/hel/verkkokauppa/merchant/model/LocaleModel.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package fi.hel.verkkokauppa.merchant.model;
```
Comment (martinldev) [inline merchantapi/src/main/java/fi/hel/verkkokauppa/merchant/model/LocaleModel.java:1]: Same comments than in ConfigurationModel.java regarding the @ToString and public fields.

File: merchantapi/src/main/java/fi/hel/verkkokauppa/merchant/model/MerchantModel.java [hunk 6]
```diff
@@ -1,14 +1,30 @@
 package fi.hel.verkkokauppa.merchant.model;
```
Comment (martinldev) [inline merchantapi/src/main/java/fi/hel/verkkokauppa/merchant/model/MerchantModel.java:1]: Same comments than in ConfigurationModel.java regarding the ToString and public fields.

File: merchantapi/src/main/java/fi/hel/verkkokauppa/merchant/model/ConfigurationModel.java (hunk 3 above)
Comment (severikupari1) [reply merchantapi/src/main/java/fi/hel/verkkokauppa/merchant/model/ConfigurationModel.java:10]: Oh good catch, I copy pasted this and just added that @Data :D

File: merchantapi/src/main/java/fi/hel/verkkokauppa/merchant/model/ConfigurationModel.java (hunk 4 above)
Comment (severikupari1) [reply merchantapi/src/main/java/fi/hel/verkkokauppa/merchant/model/ConfigurationModel.java:14]: That should be done but it differs from "common" project style so I will leave these as public fields even we have the getters and setters.

Comment (martinldev) [reply merchantapi/src/main/java/fi/hel/verkkokauppa/merchant/model/ConfigurationModel.java:14]: Yes, we can mitigate it later in a general refactor ticket https://helsinkisolutionoffice.atlassian.net/browse/KYV-658 .

**Review by synthko — APPROVED** [review]

**Pull request merged**